lib/open_trip_planner_client.ex

```diff
@@ -26,11 +27,14 @@ defmodule OpenTripPlannerClient do
     tags = if tags, do: tags, else: default_tags(params)
 
     case send_request(params) do
-      {:ok, %Plan{itineraries: itineraries}} ->
-        itineraries
+      {:ok, %QueryResult{actual_plan: actual_plan, ideal_plan: ideal_plan}} ->
+        actual_plan.itineraries
         |> Enum.map(&Map.put_new(&1, :tag, nil))
         |> ItineraryTag.apply_tags(tags)
-        |> ItineraryGroup.groups_from_itineraries(take_from_end: params.arriveBy)
+        |> ItineraryGroup.groups_from_itineraries(
+          ideal_itineraries: ideal_plan.itineraries |> ItineraryTag.apply_tags(tags),
```

Comment:
I think we can remove the `ItineraryTag.apply_tags/2` call, as we're not using tags on the ideal itineraries as far as I can tell.

Reply:
Talked offline, but realized that the `apply_tags` does in fact impact sorting, which means that it affects which trips will float to the top of the unavailable trips list.

For now, I vote we keep it, and keep fiddling as we learn more!